modules/javafx.graphics/src/main/java/com/sun/javafx/iio/ImageStorage.java

```diff
@@ -529,4 +619,26 @@ private ImageLoader getLoaderBySignature(InputStream stream, ImageLoadListener l
         // not found
         return null;
     }
+
+    /**
+     * Tries to create an {@link com.sun.javafx.iio.javax.XImageLoader} for the specified input stream.
+     * This might fail in the future if the {@code java.desktop} module is not present on the module path.
+     * At present, this will not fail because JavaFX requires the {@code java.desktop} module.
+     */
+    private synchronized ImageLoader tryCreateXImageLoader(InputStream stream) throws IOException {
+        if (xImageLoaderFactory == null) {
+            try {
+                Class<?> factoryClass = Class.forName("com.sun.javafx.iio.javax.XImageLoaderFactory");
+                xImageLoaderFactory = Optional.of((ImageLoaderFactory)factoryClass.getMethod("getInstance").invoke(null));
+            } catch (ReflectiveOperationException e) {
```

Comment:
This looks like a good approach. I simulated a failure (by adding a dependency on an unrelated module that was there at compile time, but I excluded at runtime) and it gets a `NoClassDefFoundError`.

```suggestion
            } catch (NoClassDefFoundError | ReflectiveOperationException e) {
```